packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -823,7 +823,7 @@ class _FlutterErrorDetailsNode extends DiagnosticableNode<FlutterErrorDetails> {
     if (builder != null){
       for (DiagnosticPropertiesTransformer transformer in FlutterErrorDetails.propertiesTransformers) {
         builder = DiagnosticPropertiesBuilder.fromProperties(
-          transformer(builder.properties)
+          transformer(builder.properties).toList(),
```

Comment:
nit: you are creating builders a lot more times than you need to.
I would instead only create a the new builder after completing the for loop. That would solve your concern about calling toList a ton of times.